custom/icds/forms.py

```diff
@@ -0,0 +1,141 @@
+    profile_id = forms.CharField(label=ugettext_lazy('Application Profile'),
+                                 required=False, widget=Select(choices=[]))
+    file_name = forms.CharField(label=ugettext_lazy("CCZ File Name"), required=False)
+
+    def __init__(self, request, domain, *args, **kwargs):
+        self.domain = domain
+        super(CCZHostingForm, self).__init__(*args, **kwargs)
+        self.fields['link_id'].choices = self.link_choices()
+        self.fields['app_id'].choices = self.app_id_choices()
+        self.helper = HQFormHelper()
+        if request.GET.get('app_id'):
+            self.fields['app_id'].initial = request.GET.get('app_id')
+        if request.GET.get('link_id'):
+            self.fields['link_id'].initial = request.GET.get('link_id')
+        self.helper.layout = crispy.Layout(
+            crispy.Field('link_id', css_class="hqwebapp-select2", id="link-id-select"),
+            crispy.Field('app_id', css_class="hqwebapp-select2", id='app-id-search-select'),
+            crispy.Field('version', id='version-input'),
+            crispy.Field('profile_id', id='app-profile-id-input'),
+            crispy.Field('file_name'),
+            hqcrispy.FormActions(
+                crispy.ButtonHolder(
+                    crispy.Button('search', ugettext_lazy("Search"), data_bind="click: search"),
+                    crispy.Button('clear', ugettext_lazy("Clear"), data_bind="click: clear"),
+                    Submit('submit', ugettext_lazy("Create"))
+                )
+            )
+        )
+
+    def app_id_choices(self):
+        choices = [(None, _('Select Application'))]
+        for app in get_brief_apps_in_domain(self.domain):
+            choices.append((app.id, app.name))
+        return choices
+
+    def link_choices(self):
+        choices = [(None, _('Select Link'))]
+        for link in CCZHostingLink.objects.filter(domain=self.domain):
+            choices.append((link.id, link))
+        return choices
+
+    def _version_exists(self):
+        return bool(get_version_build_id(self.domain, self.cleaned_data['app_id'],
+                                         self.cleaned_data['version']))
+
+    def clean_link_id(self):
+        link_id = self.cleaned_data.get('link_id')
+        if not link_id:
+            self.add_error('link_id', _("Please select link"))
+        return link_id
+
+    def clean_app_id(self):
+        app_id = self.cleaned_data.get('app_id')
+        if not app_id:
+            self.add_error('app_id', _("Please select application"))
+        return app_id
+
+    def clean_version(self):
+        version = self.cleaned_data.get('version')
+        if not version:
+            self.add_error('version', _("Please select version"))
+        return version
+
+    def clean_profile_id(self):
+        profile_id = self.cleaned_data.get('profile_id')
+        # ugly hack for the null added in the javascript
```

Comment:
Is this because of the null that could be a placeholder, or the null added in the "clear all options manually" section (or both)?

Reply:
this would be because of the placeholder that i had to put. I see you have suggested a way to add that without setting the null value. So if i get that working, this would not be needed anymore.